### PR TITLE
Edit food endpoint

### PR DESCRIPTION
**Pull request by tnodland**
This PR:
You can make a PATCH request to '/api/v1/foods/:id' with a name and calories in the body.  The food matching that id will be edited and the newly edited food will be returned.

A bit of ugliness in the promise inside of a promise to both update and then fetch the updated food, but when I attempted an alternative I found here: https://medium.com/@sarahdherr/sequelizes-update-method-example-included-39dfed6821d I couldn't get it to work 🤷‍♂ 

All tests are passing

### CODE REVIEW

File: test/food.spec.js [hunk 1]
```diff
@@ -52,5 +52,16 @@ describe('api', () => {
...
     })
+
+    test('It can edit a food in the database', () => {
+      return request(app).patch('/api/v1/foods/1')
+        .send({name: 'Test', calories: 100})
+        .then(response => {
+          expect(response.statusCode).toBe(201)
+          expect(response.body.id).toBe(1)
+          expect(response.body.name).toBe('Test')
+          expect(response.body.calories).toBe(100)
```
Comment (BrennanAyers) [inline test/food.spec.js]: We can edge case this more later if we want to, but this is solid coverage.

File: routes/api/v1/foods.js [hunk 2]
```diff
@@ -45,4 +45,21 @@ router.post('/', function(req, res, next) {
...
 
+router.patch('/:id', function (req, res, next) {
+ Food.update(
+   {name: req.body.name, calories: req.body.calories},
+   {where: {id: req.params.id}})
+ .then(row => {
+   Food.findOne({where: {id: req.params.id}})
+   .then(food => {
+     res.setHeader('Content-Type', 'application/json');
+     res.status(201).send(JSON.stringify(food));
```
Comment (BrennanAyers) [inline routes/api/v1/foods.js]: 👍
